api/src/functions/retrieveCheckoutSession/retrieveCheckoutSession.js

```diff
@@ -0,0 +1,31 @@
+import { logger } from 'src/lib/logger'
+import { stripe } from 'src/lib/stripe'
+
+/**
+ * The handler function is your code that processes http request events.
+ * You can use return and throw to send a response or error, respectively.
+ *
+ * Important: When deployed, a custom serverless function is an open API endpoint and
+ * is your responsibility to secure appropriately.
+ *
+ * @see {@link https://redwoodjs.com/docs/serverless-functions#security-considerations|Serverless Function Considerations}
+ * in the RedwoodJS documentation for more information.
+ *
+ * @typedef { import('aws-lambda').APIGatewayEvent } APIGatewayEvent
+ * @typedef { import('aws-lambda').Context } Context
+ * @param { APIGatewayEvent } event - an object which contains information from the invoker.
+ * @param { Context } context - contains information about the invocation,
+ * function, and execution environment.
+ */
+export const handler = async ({ body }, _context) => {
+  logger.info('Invoked retrieveCheckoutSession function')
+  const session = await stripe.checkout.sessions.retrieve(JSON.parse(body).id)
```

Comment:
This could be a Service (`api/src/services/createCheckoutSession`), and instead of making this a serverless Function, you could make it a part of your GraphQL API.